### ISSUE TITLE
Mini indutry project code

### ISSUE THREAD

**Issue by Debanjan65** [Debanjan65/college-project#2]
import pandas as pd
import random
import re
import string
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.linear_model import LogisticRegression

# 1. Generate Dataset
positive_reviews = [
    "Amazing product, works like a charm!", "Loved the fast delivery and great packaging.",
    "Fantastic quality, totally worth the price!", "Exceeded my expectations, highly recommended.",
    "Very satisfied, would definitely buy again.", "The build quality is excellent.",
    "Best purchase I've made this year!", "Absolutely love this product, five stars!",
    "Great customer support and service.", "Superb experience, very happy with my order."
]

negative_reviews = [
    "Terrible quality, broke after one use.", "Very disappointed, not as described.",
    "Poor customer service, won't buy again.", "Product arrived damaged and late.",
    "Not worth the money at all.", "Worst purchase experience ever.",
    "Completely useless, waste of money.", "Cheap material, very flimsy.",
    "Doesn't work as advertised.", "Bad experience, item was defective."
]

neutral_reviews = [
    "It's okay, does the job.", "Average quality, nothing special.",
    "Product is decent for the price.", "Neither good nor bad, just fine.",
    "Service was acceptable, not great.", "Fair product, met my basic expectations.",
    "Satisfactory, but could be better.", "Neutral feelings about this product.",
    "Okay product, arrived on time.", "Nothing exceptional, just average."
]

reviews = []
sentiments = []

for _ in range(400):
    reviews.append(random.choice(positive_reviews))
    sentiments.append(1)

for _ in range(300):
    reviews.append(random.choice(negative_reviews))
    sentiments.append(0)

for _ in range(300):
    reviews.append(random.choice(neutral_reviews))
    sentiments.append(2)

combined = list(zip(reviews, sentiments))
random.shuffle(combined)
reviews, sentiments = zip(*combined)

df = pd.DataFrame({'Review': reviews, 'Sentiment': sentiments})
df.to_csv('amazon_1000_generated_reviews.csv', index=False)

print("✅ 1000 reviews dataset saved as 'amazon_1000_generated_reviews.csv'")

# 2. Preprocessing Function
def clean_text(text):
    text = text.lower()
    text = re.sub(f"[{string.punctuation}]", "", text)
    text = re.sub(r"\d+", "", text)
    text = re.sub(r"\s+", " ", text)
    return text.strip()

df['Cleaned_Review'] = df['Review'].apply(clean_text)

# 3. Vectorization
vectorizer = TfidfVectorizer(max_features=3000, stop_words='english')
X = vectorizer.fit_transform(df['Cleaned_Review'])
y = df['Sentiment']

# 4. Train Model
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
model = LogisticRegression(max_iter=1000)
model.fit(X_train, y_train)

# 5. Predict function
def predict_sentiment(review):
    review_clean = clean_text(review)
    review_vec = vectorizer.transform([review_clean])
    prediction = model.predict(review_vec)[0]
    if prediction == 1:
        return "Positive 😊"
    elif prediction == 0:
        return "Negative 😠"
    else:
        return "Neutral 😐"

# 6. User Input Loop
print("\nType your review below (or type 'exit' to quit):\n")

while True:
    user_review = input("Enter Review: ")
    if user_review.lower() == 'exit':
        print("👋 Goodbye!")
        break
    else:
        result = predict_sentiment(user_review)
        print(f"Predicted Sentiment: {result}\n")
